server/src/main/java/org/elasticsearch/cluster/NodeUsageStatsForThreadPools.java

```diff
@@ -27,7 +27,7 @@
 public record NodeUsageStatsForThreadPools(String nodeId, Map<String, ThreadPoolUsageStats> threadPoolUsageStatsMap) implements Writeable {
 
     public NodeUsageStatsForThreadPools(StreamInput in) throws IOException {
-        this(in.readString(), in.readMap(ThreadPoolUsageStats::new));
+        this(in.readString(), in.readImmutableMap(ThreadPoolUsageStats::new));
```

Comment:
I had to change the NodeUsageStatsForThreadPool constructor to read into an immutable map because the ShardMovements*Simulator [expects and immutable map](https://github.com/elastic/elasticsearch/blob/83c1b52e0374e252853874215eb0a714a63f41df/server/src/main/java/org/elasticsearch/cluster/routing/ShardMovementWriteLoadSimulator.java#L75).